src/main/scala/scala/tools/partest/PartestDefaults.scala

```diff
@@ -25,7 +25,9 @@ object PartestDefaults {
     import scala.reflect.io.Path, Path._
     Some(Path(jdkHome) / "bin") filter (_.isDirectory) flatMap { p =>
       val candidates = (p walkFilter { e => (e.name == name || e.name.startsWith(s"$name.")) && e.jfile.canExecute }).toList
-      (candidates find (_.name == name) orElse candidates.headOption) map (_.path)
+      val plain = candidates find (_.name == name)
+      val exe = candidates find (_.name == s"${name}.exe")
+      (plain orElse exe orElse candidates.headOption).map(_.path)
```

Comment:
Is it plainer to do `val realname = if (util.Properties.isWin && !name.endsWith(".exe")) s"$name.exe" else name` and find that? Add case-insensitive test, of course. I was a little clever with the dot test to avoid these platform dances.